src/core/unit.c

```diff
@@ -6473,6 +6470,16 @@ int unit_arm_timer(
         return 0;
 }
 
+bool unit_is_filtered(Unit *u, char *const *states, char *const *patterns) {
```

Comment:
assert(u) and make u const?

Also, the function name is too vague on what precisely it does. Maybe flip the logic and name it `unit_filter_state`?

Reply:
@YHNdnzj There's stuff called by this that doesn't take a const `Unit` so can't make this change. Trying to fix that quickly spiralled out of control so I'm leaving it for another time.

Reply:
I don't find `unit_filter_state()` any clearer tbh. The dbus method is also called `ListUnitsFiltered` so `unit_is_filtered()` matches that which seems fine to me.